core/device_impl.cpp

```diff
@@ -444,12 +444,35 @@ void DeviceImpl::get_param_float_async(const std::string &name,
                                             callback));
 }
 
-MavlinkCommands::Result DeviceImpl::set_flight_mode(uint8_t custom_sub_mode, uint8_t custom_mode)
+MavlinkCommands::Result DeviceImpl::set_flight_mode(FLIGHT_MODE device_mode)
 {
-    uint8_t flag_safety_armed = DeviceImpl::is_armed() ? MAV_MODE_FLAG_SAFETY_ARMED : 0;
+    uint8_t flag_safety_armed = is_armed() ? MAV_MODE_FLAG_SAFETY_ARMED : 0;
 
     uint8_t mode = MAV_MODE_FLAG_CUSTOM_MODE_ENABLED | flag_safety_armed;
 
+    uint8_t custom_mode = px4::PX4_CUSTOM_MAIN_MODE_AUTO;
+    uint8_t custom_sub_mode = 0;
+
+    switch (device_mode) {
+        case HOLD: custom_sub_mode = px4::PX4_CUSTOM_SUB_MODE_AUTO_LOITER;
```

Comment:
Please add `custom_sub_mode = px4::PX4_CUSTOM_SUB_MODE_AUTO_LOITER;` next line for uniformity with other case values.

Reply:
Names like loiter I believe may be used by the internal state machine, but are not the public facing names.

Reply:
OK..